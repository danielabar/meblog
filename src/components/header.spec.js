import React from "react"
import renderer from "react-test-renderer"

import Header from "./header"
import NavMenu from "./nav-menu"

describe("Header", () => {
  it("renders correctly", () => {
    const testRenderer = renderer.create(<Header />)
    expect(testRenderer.toJSON()).toMatchSnapshot()

    const testInstance = testRenderer.root
    expect(testInstance.findByType(NavMenu)).toBeDefined()
  })
})
