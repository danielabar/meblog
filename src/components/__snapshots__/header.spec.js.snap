// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Header renders nav menu for wide widths 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <header
        data-testid="header"
      >
        <a
          href="/"
        >
          <div>
            <div>
              <img
                alt="Profile"
                src="/images/profile.png"
              />
            </div>
            <div
              class="undefined undefined"
            >
              Daniela Baron
            </div>
          </div>
        </a>
        <nav
          data-testid="nav-menu"
        >
          <ul>
            <li
              class="undefined undefined"
            >
              <a
                href="/blog"
              >
                Blog
              </a>
            </li>
            <li
              class="undefined undefined"
            >
              <a
                href="/about"
              >
                About
              </a>
            </li>
          </ul>
        </nav>
      </header>
    </div>
  </body>,
  "container": <div>
    <header
      data-testid="header"
    >
      <a
        href="/"
      >
        <div>
          <div>
            <img
              alt="Profile"
              src="/images/profile.png"
            />
          </div>
          <div
            class="undefined undefined"
          >
            Daniela Baron
          </div>
        </div>
      </a>
      <nav
        data-testid="nav-menu"
      >
        <ul>
          <li
            class="undefined undefined"
          >
            <a
              href="/blog"
            >
              Blog
            </a>
          </li>
          <li
            class="undefined undefined"
          >
            <a
              href="/about"
            >
              About
            </a>
          </li>
        </ul>
      </nav>
    </header>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;

exports[`Header renders nav menu responsive for narrow widths 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <header
        data-testid="header"
      >
        <a
          href="/"
        >
          <div>
            <div>
              <img
                alt="Profile"
                src="/images/profile.png"
              />
            </div>
            <div
              class="undefined undefined"
            >
              Daniela Baron
            </div>
          </div>
        </a>
        <nav
          data-testid="nav-menu-responsive"
        >
          <ul>
            <li>
              <button
                data-testid="nav-menu-responsive-menu"
              >
                <svg
                  fill="currentColor"
                  height="1.7rem"
                  stroke="currentColor"
                  stroke-width="0"
                  viewBox="0 0 24 24"
                  width="1.7rem"
                  xmlns="http://www.w3.org/2000/svg"
                >
                  <path
                    d="M3 18h18v-2H3v2zm0-5h18v-2H3v2zm0-7v2h18V6H3z"
                  />
                </svg>
                <br />
                menu
              </button>
            </li>
          </ul>
        </nav>
      </header>
    </div>
  </body>,
  "container": <div>
    <header
      data-testid="header"
    >
      <a
        href="/"
      >
        <div>
          <div>
            <img
              alt="Profile"
              src="/images/profile.png"
            />
          </div>
          <div
            class="undefined undefined"
          >
            Daniela Baron
          </div>
        </div>
      </a>
      <nav
        data-testid="nav-menu-responsive"
      >
        <ul>
          <li>
            <button
              data-testid="nav-menu-responsive-menu"
            >
              <svg
                fill="currentColor"
                height="1.7rem"
                stroke="currentColor"
                stroke-width="0"
                viewBox="0 0 24 24"
                width="1.7rem"
                xmlns="http://www.w3.org/2000/svg"
              >
                <path
                  d="M3 18h18v-2H3v2zm0-5h18v-2H3v2zm0-7v2h18V6H3z"
                />
              </svg>
              <br />
              menu
            </button>
          </li>
        </ul>
      </nav>
    </header>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
